src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -579,12 +595,15 @@ private void processLogsAndUpload() throws InterruptedException {
                     }
                     // Sort the files by the last modified time.
                     allFiles.sort(Comparator.comparingLong(File::lastModified));
-                    // If there are no rotated log files for the component, then return.
-                    if (allFiles.size() - 1 <= 0) {
-                        continue;
-                    }
-                    // Don't consider the active log file.
+
+                     // sublist allFiles to rotated file and active file, calculate the hash for active file with
+                    // some lines, default is 1 line
                     allFiles = allFiles.subList(0, allFiles.size() - 1);
+                    rotatedFiles = allFiles.subList(0, allFiles.size() - 1);
+                    File activeFile = allFiles.get(allFiles.size() - 1);
+
+                    //calculate the hash value for activeFile
```

Comment:
we're making the assumption that the active file has the latest modified time. Can we remove that assumption and still be correct?

Reply:
Good question. But will this assumption is naturally true? Because the active file must be created after the rotated file.

Reply:
Can we remove the assumption? It would be much better if we don't need to make any assumption at all.

Why do we need to treat the files differently?